### PR TITLE
[FG:InPlacePodVerticalScaling] Remove ResizePolicy defaulting

### LINKED ISSUE #128917: [FG:InPlacePodVerticalScaling] Remove ResizePolicy defaulting
/kind bug

Prior to Kubernetes v1.31, any defaulted change to the pod API could trigger running pods to restart on apiserver upgrade (fixed by https://github.com/kubernetes/kubernetes/pull/124220). Since v1.30 is still within the valid version we cannot set a default ResizePolicy.

Kubelet already handles an unset ResizePolicy as an implicit default (`NotRequired`), so all that is needed here is to remove the defaulting logic. This is probably the preferred behavior even without the version skew issue.

/sig node
/milestone v1.33
/priority important-soon
/triage accepted

### PR DESCRIPTION
**Pull request by tallclair**
#### What type of PR is this?

/kind bug
/kind api-change

#### What this PR does / why we need it:

Remove the explicit defaulting of the container ResizePolicy field, instead relying on implicit default behavior. See https://github.com/kubernetes/kubernetes/issues/128917

#### Which issue(s) this PR fixes:
Fixes https://github.com/kubernetes/kubernetes/issues/128917

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/sig node
/priority important-soon
/milestone v1.33
/triage accepted

### CODE REVIEW

File: pkg/apis/core/v1/defaults.go [hunk 1]
```diff
@@ -182,29 +182,6 @@ func SetDefaults_Pod(obj *v1.Pod) {
...
 			}
 		}
-		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) &&
-			obj.Spec.Containers[i].Resources.Requests != nil {
-			// For normal containers, set resize restart policy to default value (NotRequired), if not specified.
-			resizePolicySpecified := make(map[v1.ResourceName]bool)
-			for _, p := range obj.Spec.Containers[i].ResizePolicy {
-				resizePolicySpecified[p.ResourceName] = true
-			}
-			setDefaultResizePolicy := func(resourceName v1.ResourceName) {
```
Comment (liggitt) [inline pkg/apis/core/v1/defaults.go:192]: can drop the "defaulting" in the fuzzer as well now, right?

**Review by liggitt — COMMENTED** [review]
change lgtm

File: pkg/apis/core/v1/defaults.go (hunk 1 above)
Comment (tallclair) [reply pkg/apis/core/v1/defaults.go:192]: Yeah, but that change was rolled back in https://github.com/kubernetes/kubernetes/pull/128875. We just won't add it back when we re-enable IPPVS.

**Review by thockin — APPROVED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**